### ISSUE TITLE
Array join alias not working on distributed tables

### ISSUE THREAD

**Issue by larryluogit** [ClickHouse/ClickHouse#70851]
**Describe what's wrong**
Using an array join alias on a distributed table results in an `Unknown expression identifier` error.
https://fiddle.clickhouse.com/3ea475c1-df13-45e3-b0e9-14ba3ee27c10
This is a regression. The same query works fine in 24.3.12.75

**Does it reproduce on the most recent release?**
Yes. 
The issue can be reproduced in 24.9.2.42
https://fiddle.clickhouse.com/3ea475c1-df13-45e3-b0e9-14ba3ee27c10

**How to reproduce**

```
select version();
CREATE TABLE default.arrays_test
(
    s String,
    `custom_tags.key` Array(String),
    `custom_tags.value` Array(String)
) 
ENGINE =MergeTree()
ORDER BY (s)
SETTINGS index_granularity = 8192;

CREATE TABLE IF NOT EXISTS default.arrays_test_distributed
    AS default.arrays_test
ENGINE = Distributed('default', 'default', 'arrays_test', rand());


INSERT INTO default.arrays_test_distributed
VALUES ('Hello', ['k1'],['1']), ('World', ['k2'],['2']), ('Goodbye', ['k3'],['3']);

SELECT s
FROM default.arrays_test_distributed
--FROM default.arrays_test
ARRAY JOIN custom_tags AS array_join_alias
WHERE (array_join_alias.key = 'k1')


Received exception from server (version 24.9.2):
Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Unknown expression identifier `array_join_alias` in scope SELECT __table2.s AS s FROM default.arrays_test AS __table2 ARRAY JOIN array_join_alias AS array_join_alias WHERE getSubcolumn(array_join_alias, 'key') = 'k1'. Maybe you meant: ['array_join_alias']. (UNKNOWN_IDENTIFIER)
(query: SELECT s
FROM default.arrays_test_distributed
--FROM default.arrays_test
ARRAY JOIN custom_tags AS array_join_alias
WHERE (array_join_alias.key = 'k1'))

```
**Expected behavior**
The provided query executes successfully

**Additional context**
The query runs fine when using local table.

**Comment by SaltTan** [comment]
works with `enable_analyzer=0`
https://fiddle.clickhouse.com/554ba19b-1747-4a42-ba93-ea66085c1613

**Comment by larryluogit** [comment]
Can this issue get some attention, please? It is preventing users from upgrading to 24.8 LTS.

**Comment by rschu1ze** [comment]
Fixed in `master` as per #72608 (I verified).

**Comment by larryluogit** [comment]
The change was backported to 24.8 https://github.com/ClickHouse/ClickHouse/pull/72926. 
We will verify it. Thank you @rschu1ze
